test build warnings: Interop\MarshalAPI\UnsafeAddrOfPinnedArrayElement\UnsafeAddrOfPinnedArrayElement.csproj

```
  3180>UnsafeAddrOfPinnedArrayElement.cs(16,52): warning CS0618: 'Marshal.UnsafeAddrOfPinnedArrayElement(Array, int)' is obsolete: 'UnsafeAddrOfPinnedArrayElement(Array, Int32) may be unavailable in future releases. Instead, use UnsafeAddrOfPinnedArrayElement<T>(T[], Int32). For more info, go to http://go.microsoft.com/fwlink/?LinkID=296517' [D:\gh\coreclr\tests\src\Interop\MarshalAPI\UnsafeAddrOfPinnedArrayElement\UnsafeAddrOfPinnedArrayElement.csproj]
```

--- a/tests/src/Interop/MarshalAPI/GetNativeVariantForObject/GetNativeVariantForObject.csproj
+++ b/tests/src/Interop/MarshalAPI/GetNativeVariantForObject/GetNativeVariantForObject.csproj
@@ -16,6 +16,7 @@
 
     <NuGetPackageImportStamp>7a9bfb7d</NuGetPackageImportStamp>
     <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+    <NoWarn>0618;0649</NoWarn>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">

--- a/tests/src/Interop/MarshalAPI/Miscellaneous/MarshalClassTests.cs
+++ b/tests/src/Interop/MarshalAPI/Miscellaneous/MarshalClassTests.cs
@@ -56,9 +56,7 @@ class MarshalClassTests
         int retVal = 100;
 
         IntPtr ip;
-        Object o;
         SomeTestStruct someTs = new SomeTestStruct();
-        StructWithFxdLPSTRSAFld someTs_FxdLPSTR = new StructWithFxdLPSTRSAFld();
 
 #if BUG_876976
         Console.WriteLine("Testing SizeOf...");
@@ -308,7 +306,7 @@ class MarshalClassTests
         Console.WriteLine("\n\tPassing structure that has no layout and CANNOT be marshaled");
         try
         {
-            Marshal.SizeOf(someTs_FxdLPSTR.GetType());
+            Marshal.SizeOf(typeof(StructWithFxdLPSTRSAFld));
             retVal = 0;
             Console.WriteLine("\t\tNO EXCEPTION THROWN! FAILED!");
         }
@@ -371,7 +369,7 @@ class MarshalClassTests
         Console.WriteLine("\n\tPassing structure that has no layout and CANNOT be marshaled");
         try
         {
-            Marshal.OffsetOf(someTs_FxdLPSTR.GetType(), "Arr");
+            Marshal.OffsetOf(typeof(StructWithFxdLPSTRSAFld), "Arr");
             retVal = 0;
             Console.WriteLine("\t\tNO EXCEPTION THROWN! FAILED!");
         }
@@ -493,6 +491,7 @@ class MarshalClassTests
 
         Console.WriteLine("DONE testing Copy.");
         
+#if ISSUE_6605
         //////////////////////////////////////////////////////////////
         //IsComObject
         /////////////////////////////////////////////////////////////
@@ -513,9 +512,9 @@ class MarshalClassTests
             retVal = 0;
             Console.WriteLine("\t\tUNEXPECTED EXCEPTION:\n\t\t\t" + e.ToString());
         }
-
+#endif
         Console.WriteLine("DONE testing IsComObject.");
 
-        return 100;
+        return retVal;
     }
 }

--- a/tests/src/Interop/MarshalAPI/Miscellaneous/MarshalClassTests.csproj
+++ b/tests/src/Interop/MarshalAPI/Miscellaneous/MarshalClassTests.csproj
@@ -16,6 +16,7 @@
 
     <NuGetPackageImportStamp>7a9bfb7d</NuGetPackageImportStamp>
     <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+    <NoWarn>0618;0649</NoWarn>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">

--- a/tests/src/Interop/MarshalAPI/OffsetOf/OffsetOf.csproj
+++ b/tests/src/Interop/MarshalAPI/OffsetOf/OffsetOf.csproj
@@ -16,6 +16,7 @@
 
     <NuGetPackageImportStamp>7a9bfb7d</NuGetPackageImportStamp>
     <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+    <NoWarn>0618;0649;0169</NoWarn>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">

--- a/tests/src/Interop/MarshalAPI/UnsafeAddrOfPinnedArrayElement/UnsafeAddrOfPinnedArrayElement.csproj
+++ b/tests/src/Interop/MarshalAPI/UnsafeAddrOfPinnedArrayElement/UnsafeAddrOfPinnedArrayElement.csproj
@@ -16,6 +16,7 @@
 
     <NuGetPackageImportStamp>7a9bfb7d</NuGetPackageImportStamp>
     <DefineConstants>$(DefineConstants);STATIC</DefineConstants>
+    <NoWarn>0618</NoWarn>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">

